packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/ExpectedType.kt

```diff
@@ -96,5 +114,40 @@ class ExpectedType(
     fun forMap(valueType: ExpectedType) = ExpectedType(
       SingleType(CppType.MAP, arrayOf(valueType))
     )
+
+    fun fromKType(type: KType): ExpectedType {
+      val kClass = type.classifier as? KClass<*> ?: throw IllegalArgumentException("...")
+      if (Int::class == kClass) {
+        return ExpectedType(SingleType(CppType.INT))
+      }
+      if (Long::class == kClass) {
+        return ExpectedType(SingleType(CppType.LONG))
+      }
+      if (Double::class == kClass) {
+        return ExpectedType(SingleType(CppType.DOUBLE))
+      }
+      if (Float::class == kClass) {
+        return ExpectedType(SingleType(CppType.FLOAT))
+      }
+      if (Boolean::class == kClass) {
+        return ExpectedType(SingleType(CppType.BOOLEAN))
+      }
+      if (String::class == kClass) {
+        return ExpectedType(SingleType(CppType.STRING))
+      }
```

Comment:
```suggestion
      when (kClass) {
        Int::class -> return ExpectedType(SingleType(CppType.INT))
        Long::class -> return ExpectedType(SingleType(CppType.LONG))
        Double::class -> return ExpectedType(SingleType(CppType.DOUBLE))
        Float::class -> return ExpectedType(SingleType(CppType.FLOAT))
        Boolean::class -> return ExpectedType(SingleType(CppType.BOOLEAN))
        String::class -> return ExpectedType(SingleType(CppType.STRING))
      }
```
I think, it looks better